core/src/test/scala/integration/kafka/api/ConsumerBounceTest.scala

```diff
@@ -147,6 +148,230 @@ class ConsumerBounceTest extends IntegrationTestHarness with Logging {
     }
   }
 
+  /**
+   * Consumer is closed while cluster is healthy. Consumer should complete pending offset commits
+   * and leave group. New consumer instance should be able join group and start consuming from
+   * last committed offset.
+   */
+  @Test
+  def testClose() {
```

Comment:
I have removed `testCloseWithRandomBrokerFailures` and refactored the other tests to run more logic concurrently. Tests for different options in the same scenario are now run concurrently. The individual cases were running within the time expected, but the tests were taking a long time because it takes time to create each scenario, `close()` and then cleanup. The code is slightly more complicated now because of concurrency, but the total time for the tests has come down to 1m42s (1m4s for the existing tests in the class and 38s for the new close tests).